src/utils/token_details.py

```diff
@@ -0,0 +1,31 @@
+"""
+Very basic Token Info Fetcher that gets token decimals
+"""
+import functools
+import logging.config
+from duneapi.types import Address
+from web3 import Web3
+
+from src.constants import ERC20_ABI, NODE_URL
+
+
+log = logging.getLogger(__name__)
+logging.config.fileConfig(fname="logging.conf", disable_existing_loggers=False)
+
+
+@functools.cache
+def get_token_decimals(address: str | Address) -> int:
```

Comment:
Would it make sense to to have this function part of the class `Token`?  then one could write token.get_decimals(), which seems easier

Reply:
I like this idea and will do it.

Reply:
Oh yes, I remember why. Because this method requires a valid Web3 connection. If we put this into the class, then even when we don't need it it will expect a web3 connection. Generally the main run of this program doesn't ever need to fetch decimals. I could probably work it in a way that doesn't need this, but it seems like a lot of effort for not much bonus.